app/src/androidTestE2eTest/java/io/appium/uiautomator2/unittest/test/DeviceCommandsTest.java

```diff
@@ -0,0 +1,449 @@
+                contextId);
+        response = getText(response.getElementId());
+        assertEquals("Animation", response.getValue());
+
+        //parent element - By.xpath
+        response = findElement(By.xpath("//hierarchy//*[@class='android.widget.FrameLayout'][2]"));
+        contextId = response.getElementId();
+
+        //child element - By.xpath  (UiObject2)
+        response = findElement(By.xpath("//*[@class='android.widget.TextView'][2]"), contextId);
+        response = getText(response.getElementId());
+        assertEquals("Accessibility", response.getValue());
+
+        //parent element - By.androidUiAutomator (UiObject)
+        response = findElement(By.androidUiAutomator("new UiSelector()"
+                + ".resourceId(\"android:id/list\");"));
+        contextId = response.getElementId();
+
+        //child element - By.xpath  (UiObject2)
+        response = findElement(By.xpath("//*[@class='android.widget.TextView'][2]"), contextId);
+        response = getText(response.getElementId());
+        assertEquals("Accessibility", response.getValue());
+
+        response = findElement(By.accessibilityId("Animation"));
+        clickAndWaitForStaleness(response.getElementId());
+        response = findElement(By.accessibilityId("Events"));
+        clickAndWaitForStaleness(response.getElementId());
+
+        //parent element - By.xpath (UiObject2)
+        response = findElement(By.xpath("//*[@class='android.widget.LinearLayout'][3]"));
+        contextId = response.getElementId();
+
+        //child  element - By.androidUiAutomator (UiObject)
+        response = findElement(By.androidUiAutomator("new UiSelector().className(\"android.widget" +
+                ".TextView\")"), contextId);
+        response = getText(response.getElementId());
+        assertEquals("Animator Events:   ", response.getValue());
+    }
+
+    @Test
+    public void findElementWithAttributes() throws JSONException {
+        scrollTo("Views");
+        Response response = findElement(By.accessibilityId("Views"));
+        clickAndWaitForStaleness(response.getElementId());
+
+        By by = By.accessibilityId("Buttons");
+        response = findElement(by);
+        String elementId = response.getElementId();
+        assertEquals(by + " should be found", WDStatus.SUCCESS.code(), response.getStatus());
+
+        response = getAttribute(elementId, "clickable");
+        assertEquals("true", response.getValue());
+
+        response = getAttribute(elementId, "enabled");
+        assertEquals("true", response.getValue());
+    }
+
+    @Test
+    public void findElementWithClassName() {
+        Response response = findElement(By.className("android.widget.TextView"));
+        response = getText(response.getElementId());
+        assertEquals("API Demos", response.getValue());
+    }
+
+    @Test
+    public void findElementWithIndex() {
+        Response response = findElement(By.xpath("//*[@resource-id='android:id/list']" +
```

Comment:
I assume that using private `index` field in requests is a bad practice

Reply:
Should we delete this tests?

Reply:
yep do it